packages/event-bus-redis/src/services/event-bus-redis.ts

```diff
@@ -0,0 +1,197 @@
+import {
+  AbstractEventBusModuleService,
+  ConfigModule,
+  ConfigurableModuleDeclaration,
+  Logger,
+  MODULE_RESOURCE_TYPE
+} from "@medusajs/medusa"
+import { ConnectionOptions, Queue, Worker } from "bullmq"
+import Redis from "ioredis"
+import { MedusaError } from "medusa-core-utils"
+import { BullJob, EmitOptions, EventBusRedisModuleOptions } from "../types"
+
+type InjectedDependencies = {
+  logger: Logger
+  configModule: ConfigModule
+}
+
+/**
+ * Can keep track of multiple subscribers to different events and run the
+ * subscribers when events happen. Events will run asynchronously.
+ */
+export default class RedisEventBusService extends AbstractEventBusModuleService {
+  protected readonly config_: ConfigModule
+  protected readonly logger_: Logger
+  protected readonly moduleOptions_: EventBusRedisModuleOptions
+  protected readonly moduleDeclaration_: ConfigurableModuleDeclaration
+
+  protected queue_: Queue
+
+  constructor(
+    { configModule, logger }: InjectedDependencies,
+    moduleOptions: EventBusRedisModuleOptions = {},
+    moduleDeclaration: ConfigurableModuleDeclaration,
+  ) {
+    // @ts-ignore
+    super(...arguments)
+
+    if (moduleDeclaration?.resources !== MODULE_RESOURCE_TYPE.SHARED) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        "At the moment this module can only be used with shared resources"
+      )
+    }
+
+    this.moduleOptions_ = moduleOptions
+    this.moduleDeclaration_ = moduleDeclaration
+    this.config_ = configModule
+    this.logger_ = logger
+
+    const connection = this.connect()
```

Comment:
non-blocking/suggestion: that connection could be moved to `loader/index.ts` and actually wait for the connection before use it.

Reply:
Yeah, I like that. I've added. that in 0b2dc23